tests/integration/shell/matcher.py

```diff
@@ -159,6 +161,11 @@ def test_grain(self):
         data = '\n'.join(data)
         self.assertIn('sub_minion', data)
         self.assertNotIn('minion', data.replace('sub_minion', 'stub'))
+        # Custom grain
+        data = self.run_salt('-t 1 -G "match:maker" test.ping')
```

Comment:
Where does the `{'match': 'maker'}` grain come from? I think that is why this test is failing. There isn't a return from the minions on this custom grain call. The first `assertIn` is passing because 'minion' is found in `['No minions matched the target. No command was sent, no jid was assigned.']`.

You could maybe use the custom grains files in `integration/files/file/base/_grains/*` for this test instead.